src/chrome/content/toolbar_button.js

```diff
@@ -163,14 +148,75 @@ httpsEverywhere.toolbarButton = {
       return window.BrowserApp.selectedBrowser;
     }
   },
+   
+  /**
+   * Update the toolbar button image and menuitem checkboxes to current settings.
+   * This is used when the UI is initialized.
+   */
+  updateMenuItems: function() {
+    HTTPSEverywhere.log(INFO, 'Updating toolbar button image and menu item checkboxes.');
+    var tb = httpsEverywhere.toolbarButton;
 
+    var counterItem = document.getElementById('https-everywhere-counter-item');
+    
+    var uiFound = false;
+    
+    if (counterItem) { // If we can find the counterItem like this, we can find everything in the UI.
+      uiFound = true;
+      var counterTotalItem = document.getElementById('https-everywhere-counter-total-item');
+      var httpNowhereItem = document.getElementById('http-nowhere-item');
+      var toolbarbutton = document.getElementById('https-everywhere-button');
+    } else {
+      // If the toolbar button is hidden away in the Firefox hamburger menu,
+      // we won't find it by the above, and need to look somewhere else
+      let palette = document.getElementById("navigator-toolbox").palette;
```

Comment:
I suspect this can return undefined, or may in the future if Firefox changes its XUL. We should check the result of getElementById before dereferencing .palette.